android/anc/src/main/java/org/smartregister/fhircore/anc/ui/report/ReportHomeActivity.kt

```diff
@@ -17,29 +17,344 @@
     val repository = ReportRepository((application as AncApplication).fhirEngine, patientId)
-    val viewModel = ReportViewModel.get(this, application as AncApplication, repository)
-    viewModel.backPress.observe(
+    val dispatcher: DispatcherProvider = DefaultDispatcherProvider
+    reportViewModel = ReportViewModel(repository, dispatcher)
+    reportViewModel.backPress.observe(
       this,
       {
         if (it) {
           finish()
         }
       }
     )
-    setContent { AppTheme { ReportHomeScreen(viewModel) } }
+
+    fileUtil = FileUtil()
+    cqlBaseURL =
+      this.let { fileUtil.getProperty("smart_register_base_url", it, cqlConfigFileName) }!!
+
+    libraryURL =
+      cqlBaseURL + this.let { fileUtil.getProperty("cql_library_url", it, cqlConfigFileName) }
+
+    cqlHelperURL =
+      cqlBaseURL +
+        this.let { fileUtil.getProperty("cql_helper_library_url", it, cqlConfigFileName) }
+
+    valueSetURL =
+      cqlBaseURL + this.let { fileUtil.getProperty("cql_value_set_url", it, cqlConfigFileName) }
+
+    patientURL =
+      cqlBaseURL + this.let { fileUtil.getProperty("cql_patient_url", it, cqlConfigFileName) }
+
+    measureEvaluateLibraryURL =
+      this.let {
+        fileUtil.getProperty("cql_measure_report_library_value_sets_url", it, cqlConfigFileName)
+      }!!
+
+    measureTypeURL =
+      this.let { fileUtil.getProperty("cql_measure_report_resource_url", it, cqlConfigFileName) }!!
+
+    cqlMeasureReportURL =
+      this.let { fileUtil.getProperty("cql_measure_report_url", it, cqlConfigFileName) }!!
+
+    cqlMeasureReportLibInitialString =
+      this.let {
+        fileUtil.getProperty("cql_measure_report_lib_initial_string", it, cqlConfigFileName)
+      }!!
+
+    setContent {
+      AppTheme {
+        Surface(color = colorResource(id = R.color.white)) {
+          Column {
+            val circularProgressBarVisible = remember { mutableStateOf(value = true) }
+
+            TopAppBar(
+              title = {
+                Text(text = stringResource(id = R.string.reports), Modifier.testTag(TOOLBAR_TITLE))
+              },
+              navigationIcon = {
+                IconButton(
+                  onClick = reportViewModel::onBackPress,
+                  Modifier.testTag(TOOLBAR_BACK_ARROW)
+                ) { Icon(Icons.Filled.ArrowBack, contentDescription = "Back arrow") }
+              }
+            )
+            if (circularProgressBarVisible.value) {
+              CircularProgressBarDemo()
+            }
+            executor.execute {
+              loadCQLLibraryData()
+              loadMeasureEvaluateLibrary()
+              handler.post { circularProgressBarVisible.value = false }
+            }
+            ReportHomeScreen(reportViewModel)
+          }
+        }
+      }
+    }
+  }
+
+  fun loadCQLLibraryData() {
+    dir = File(this.filesDir, "$dirCQLDirRoot/$fileNameMainLibraryCql")
```

Comment:
suggestion : such logic can be move to viewmodel via passing dir and observable fun.

Reply:
@AbdulWahabMemon so if you look in the lines below is actually referencing the ViewModel to loaddata from server